File: Javascript/Bit Manipulation/Number-Of-1-Bits.js
/**
 *
 * @param {number} n
 * @returns {number}
 */
const hammingWeight = (n) => {
  let result = 0;

  /* 
    1100 = 12    result = 0;
    1011 = 11

    1100 & 1011 = 1000 = 8  result = 1
    0111 = 7

    1000 & 0111 = 0000 = 0 result = 2

  */

  while (n != 0) {
    n &= n - 1;
    result++;
  }

  return result;
};

File: Rust/Bit Manipulation/Number-Of-1-Bits.rs
pub struct Solution {}

impl Solution {
    pub fn hammingWeight(n: u32) -> i32 {
        let mut result = 0;
        let mut n = n;

        for _ in 0..32 {
            result += (n % 2) as i32;
            n >>= 1;
        }

        result

        //  n.count_ones() as i32
    }
}

fn main() {}
